METRON-1441: Create complementary Solr schemas for the main sensors

cestella opened the pull request:
## Contributor Comments
We have ES templates for bro, snort, yaf, and error, we need corresponding solr schemas for these collections.

Right now this is tested via the SchemaTranslatorTest, which spins up the current version of Solr loads the schemas and writes sample data and ensures they come back properly. 

Note: This is an intermediate move to getting full solr support.  These schemas will be refined more, likely.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

ottobackwards commented on metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java:
```diff
@@ -0,0 +1,188 @@
+public class SchemaTranslatorTest {
+
+  /**
+{"adapter.threatinteladapter.end.ts":"1517499201357","bro_timestamp":"1517499194.7338","ip_dst_port":8080,"enrichmentsplitterbolt.splitter.end.ts":"1517499201202","enrichmentsplitterbolt.splitter.begin.ts":"1517499201200","adapter.hostfromjsonlistadapter.end.ts":"1517499201207","adapter.geoadapter.begin.ts":"1517499201209","uid":"CUrRne3iLIxXavQtci","trans_depth":143,"protocol":"http","original_s ...
```
Can we make these files please?  Maybe add a new directory "enriched" under the sample data dir in the integration testing module

cestella replied:
We could, though this test probably won't last past the feature branch.

ottobackwards replied:
I wouldn't be so sure.  When we get serious about ES templates and these schemas they will be tested per parser.   As part of developing a parser you can validate your schemas etc like this.

cestella replied:
Alright, you got it then :)

ottobackwards commented on metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java:
```diff
@@ -0,0 +1,188 @@
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.*;
+
```
Shouldn't this be an integration test?

cestella replied:
Yes, I absolutely can.

cestella commented:
@ottobackwards Very likely these schema files won't stay in this spot, but the final resting spot won't be apparent until we figure out how to automatically apply the schemas.  Treat this PR as just unlocking progress for downstream PRs (like correcting SolrWriter to write to Solr again).

cestella commented:
Ok, I did the following:
* Augmented the readme to point to the Solr documentation around schemas.  Keep in mind, this is intermediate work that will feed into the "install Solr" work
* Added yaf and error schemas
* Renamed the test to an integration test
* Moved the data from multiline to separate files

merrimanr commented:
I tested this in full dev using the install script in https://github.com/apache/metron/pull/918.  I was able to create collections for each schema except for "error".  For that to work properly, I had to: 

- remove `docValues="true"` from the "bytes" field type
- add the "guid" field used in other schemas

Still working on indexing data into these collections but so far so good.

cestella commented:
@merrimanr I'm confused about the guid field.  I had to take that out given the sample data that I generated from full-dev (they didn't have a `guid` field).  How did you create the errors?

merrimanr commented:
The errors I found happened when I tried to create collections with each schema.  The error related to the `guid` field happens because it is defined as the unique key but not included in the list of fields (for error schema).

cestella commented:
@merrimanr Ah!  Yes, we really should set a `guid` field for errors.

cestella commented:
@merrimanr Things should be fixed at this point.  I'm going to spin it up and validate things this morning.

ottobackwards commented:
So, the idea is that we have a 'core' for each parser?  and for errors?

merrimanr commented:
I would say collection for each parser and an error collection.  It's similar to ES templates where we define fields, types, etc.

cestella commented:
Ugh, sorry, missed the guid field.  *now* it's ready to test with guid field set up.

merrimanr commented:
Looks good to me.  +1

simonellistonball commented:
Should we tidy up the ordering of the schema files for better legibility (uniquekey next to the field, dynamic catch alls in a consistent location, some semantic ordering of the key elements)?

cestella commented:
@simonellistonball Yes, we should.  I added the relevant context and grouping for each of the schemas that we ship by default.  Bro has more context as there were more comments in the ES schema.  I brought over whatever comments were relevant and grouped appropriately and as best I could in the others.

Let me know if I missed anything.

simonellistonball commented on metron-platform/metron-solr/src/main/config/schema/error/solrconfig.xml:
```diff
@@ -0,0 +1,1601 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Is this just basic stock solr? The /browse endpoint configs seem like they come from the tutorial sample, and could be mis-leading for example.

simonellistonball commented:
@cestella much neater. Thank you! I'll put my data schema OCD away now.
